### ISSUE TITLE
Executing `Tree._gui_input` crashes Godot

### ISSUE THREAD

**Issue by qarmin** [godotengine/godot#70097]
### Godot version

3.6.beta.custom_build. 71f5bb391

### System information

Ubuntu 22.10 - AMD RX 470, Gnome shell 43 X11

### Issue description

When executing
```
extends Node
func _process(delta):

	var temp_variable376 = Tree.new()
	add_child(temp_variable376)
	temp_variable376._gui_input(null)
```
Godot crashes:
```
scene/gui/tree.cpp:2181:24: runtime error: member call on null pointer of type 'struct InputEvent'
scene/gui/tree.cpp:2181:24: runtime error: member access within null pointer of type 'struct InputEvent'
AddressSanitizer:DEADLYSIGNAL
=================================================================
==20435==ERROR: AddressSanitizer: SEGV on unknown address 0x000000000000 (pc 0x000006cb11af bp 0x7ffea81ce600 sp 0x7ffea81cc240 T0)
==20435==The signal is caused by a READ memory access.
==20435==Hint: address points to the zero page.
    #0 0x6cb11af in Tree::_gui_input(Ref<InputEvent>) scene/gui/tree.cpp:2181
    #1 0x5930da6 in MethodBind1<Ref<InputEvent> >::call(Object*, Variant const**, int, Variant::CallError&) core/method_bind.gen.inc:759
    #2 0x112ba50c in Object::call(StringName const&, Variant const**, int, Variant::CallError&) core/object.cpp:918
    #3 0x11741bce in Variant::call_ptr(StringName const&, Variant const**, int, Variant*, Variant::CallError&) core/variant_call.cpp:1238
    #4 0x1b3626e in GDScriptFunction::call(GDScriptInstance*, Variant const**, int, Variant::CallError&, GDScriptFunction::CallState*) modules/gdscript/gdscript_function.cpp:1050
    #5 0x19a7570 in GDScriptInstance::call_multilevel(StringName const&, Variant const**, int) modules/gdscript/gdscript.cpp:1211
    #6 0x53e1f6b in Node::_notification(int) scene/main/node.cpp:58
    #7 0x6e244e in Node::_notificationv(int, bool) (/home/runner/work/Qarminer/Qarminer/test_project+0x6e244e)
    #8 0x112bab85 in Object::notification(int, bool) core/object.cpp:927
    #9 0x55c33a0 in SceneTree::_notify_group_pause(StringName const&, int) scene/main/scene_tree.cpp:1133
    #10 0x55acb00 in SceneTree::idle(float) scene/main/scene_tree.cpp:636
    #11 0x6a57c9 in Main::iteration() main/main.cpp:2340
    #12 0x52b78d in OS_X11::run() platform/x11/os_x11.cpp:4035
    #13 0x472d61 in main platform/x11/godot_x11.cpp:59
    #14 0x7fa1d73f9d8f  (/lib/x86_64-linux-gnu/libc.so.6+0x29d8f)
    #15 0x7fa1d73f9e3f in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x29e3f)
    #16 0x472934 in _start (/home/runner/work/Qarminer/Qarminer/test_project+0x472934)

AddressSanitizer can not provide additional info.
SUMMARY: AddressSanitizer: SEGV scene/gui/tree.cpp:2181 in Tree::_gui_input(Ref<InputEvent>)
==20435==ABORTING
```

This example was found by Godot fuzzer - [Qarminer](https://github.com/qarmin/Qarminer), so it is quite unlikelly that this code could be used in real project, but still this should be handled gracefully.

Memory leaks or asan backtraces are visibe when using Godot build with sanitizers support - https://github.com/qarmin/GodotBuilds/actions (linux -> linux-editor-sanitizers)

### Steps to reproduce

Above

### Minimal reproduction project

N/A

**Comment by timothyqiu** [comment]
Kind of similar to #69214

**Comment by akien-mga** [comment]
Fixed by #70111.
